lib/web_of_science/links_client.rb

```diff
@@ -0,0 +1,85 @@
+module WebOfScience
+  # Links AMR (Article Match Retrieval) Service
+  # @see http://ipscience-help.thomsonreuters.com/LAMRService/WebServicesOverviewGroup/overview.html Service documentation
+  class LinksClient
+
+    LINKS_HOST = 'https://ws.isiknowledge.com'.freeze
+    LINKS_PATH = '/cps/xrpc'.freeze
+
+    attr_reader :username, :password, :host
+
+    # @param [String] username
+    # @param [String] password
+    # @param [String] host
+    def initialize(username: nil, password: nil, host: LINKS_HOST)
+      @host = host
+      @username = username
+      @password = password
+      (@username, @password) = Base64.decode64(Settings.WOS.AUTH_CODE).split(':', 2) unless username
+    end
+
+    # Retrieve identifier 'fields' for the record 'ids'
+    # @param [Array<String>] ids
+    # @param [Array<String>] fields (defaults to ['ut', 'doi', 'pmid'])
+    # @return [Hash<String => Hash>]
+    def links(ids, fields = %w(ut doi pmid))
+      ids.uniq!
+      raise ArgumentError, '1-50 ids required' if ids.empty? || ids.count > 50
+      raise ArgumentError, 'fields cannot be empty' if fields.empty?
+      response = connection.post do |req|
+        req.path = LINKS_PATH
+        req.body = request_body(ids, fields)
+      end
+      response_parse(response)
+    end
+
+    private
+
+      # @return [Faraday::Connection]
+      def connection
+        @connection ||= Faraday.new(url: host) do |faraday|
+          faraday.use Faraday::Response::RaiseError
+          faraday.adapter :httpclient
+        end
+      end
+
+      # @return [ActionView::Base] used to render as though in an rails controller
+      def renderer
+        @renderer ||= begin
+          view_paths = ActionView::PathSet.new([File.dirname(__FILE__)])
+          ActionView::Base.new(view_paths, {})
+        end
+      end
+
+      # @param [Array<String>] fields
+      # @return [String]
+      def request_body(ids, fields)
+        renderer.render(
+          file: 'links_request.xml',
```

Comment:
Is this a reference to a file in this codebase?

Reply:
Yes.  It is moved in this PR.

Reply:
This file is in the same directory as this class in this PR.  The render paths can only traverse this directory in this PR.

Reply:
OK. Mostly I was wondering whether it would make sense to freeze this string as a constant (`TEMPLATE_PATH` or some such?), but that's not that big a deal. Thanks, you two.